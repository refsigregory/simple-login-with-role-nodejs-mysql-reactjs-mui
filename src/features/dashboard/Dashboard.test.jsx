import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import Dashboard from './Dashboard';

const testExample = <Dashboard />;
const testRouter  = 
<Router>
  {testExample}
</Router>;

it('renders correctly', () => {
  const tree = renderer
    .create(testRouter)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
