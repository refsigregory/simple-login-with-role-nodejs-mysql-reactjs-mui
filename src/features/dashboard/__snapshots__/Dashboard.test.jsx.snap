// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiBox-root css-k008qs"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionAbsolute css-n7rtur-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-m9mp0w-MuiToolbar-root"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-1johsky-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </button>
      <h1
        className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-uh02xs-MuiTypography-root"
      >
        Dashboard
      </h1>
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="LogoutIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m17 7-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4V5z"
          />
        </svg>
      </button>
    </div>
  </header>
  <div
    className="MuiDrawer-root MuiDrawer-docked css-cvumne-MuiDrawer-docked"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
    >
      <div
        className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1dd0suk-MuiToolbar-root"
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ChevronLeftIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </button>
      </div>
      <hr
        className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
      <nav
        className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
      >
        <a
          className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1n8yj6x-MuiTypography-root-MuiLink-root"
          href="/dashboard"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <div
            className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DashboardIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z"
                />
              </svg>
            </div>
            <div
              className="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
              >
                Dashboard
              </span>
            </div>
          </div>
        </a>
      </nav>
    </div>
  </div>
  <main
    className="MuiBox-root css-fxbtpg"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    />
    <div
      className="MuiContainer-root MuiContainer-maxWidthLg css-1f2kw6w-MuiContainer-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8 MuiGrid-grid-lg-9 css-1dt1sxv-MuiGrid-root"
        >
          <div
            className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-t5ouwv-MuiPaper-root"
          />
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 css-3p2wky-MuiGrid-root"
        >
          <div
            className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-t5ouwv-MuiPaper-root"
          />
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1slnj5u-MuiPaper-root"
          />
        </div>
      </div>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-9wziol-MuiTypography-root"
      >
        Copyright Â© 
        <a
          className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1cghntr-MuiTypography-root-MuiLink-root"
          href="https://ref.si/"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          My Website
        </a>
         
        2022
        .
      </p>
    </div>
  </main>
</div>
`;
